plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureRepositoryPlugin.java

```diff
@@ -40,6 +43,15 @@
  */
 public class AzureRepositoryPlugin extends Plugin implements RepositoryPlugin, ReloadablePlugin {
 
+    public static final String REPOSITORY_THREAD_POOL_NAME = "azure-repository";
+
+    private static final Setting<Integer> THREADPOOL_SIZE_SETTING = Setting.intSetting(
```

Comment:
Isn't this already configurable through the standard threadpool settings (see https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-threadpool.html)? Why does it need an extra setting here?

Reply:
My bad didn't know about the mechanics of this :(